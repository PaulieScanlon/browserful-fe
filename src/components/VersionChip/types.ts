export enum overrideTypes {
  IS_INCLUDED = 'isIncluded',
  IS_EXCLUDED = 'isExcluded'
}
