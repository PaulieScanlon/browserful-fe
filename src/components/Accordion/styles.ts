import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  maxHeight: string;
}

export const AccordionWrapper = styled.div({
  label: 'acorrdion-wrapper',
  position: 'relative',
  width: '100%',
  boxShadow: common.materialBoxShadow1
});

export const Input = styled.input<IProps>(
  {
    label: 'input',
    position: 'absolute',
    top: '-10px',
    left: '-10px',
    display: 'none',

    ':checked + label': {
      '&:after': {
        content: `""`,
        width: '14px',
        height: '14px',
        borderWidth: '4px',
        borderColor: `${colours.pink}`,
        backgroundColor: `${colours.white}`,
        color: colours.pink
      }
    }
  },
  ({ maxHeight }) => ({
    ':checked ~ div': {
      maxHeight: maxHeight
    }
  })
);

export const Label = styled.label({
  label: 'label',
  position: 'relative',
  display: 'block',
  backgroundColor: 'white',
  cursor: 'pointer',
  padding: `${spaceMd}px`,
  color: colours.greyDark,
  borderBottom: `1px solid ${colours.greyLight}`,
  transition: `${common.transition}`,
  boxSizing: 'border-box',
  ':after': {
    content: `""`,
    position: 'absolute',
    width: '20px',
    height: '20px',
    borderWidth: '1px',
    borderStyle: 'solid',
    borderRadius: '100%',
    right: `${spaceMd}px`,
    textAlign: 'center',
    borderColor: `${colours.greyLight}`,
    transition: `${common.transition}`,
    color: colours.greyLight
  }
});

export const ContentWrapper = styled.div({
  label: 'content-wrapper',
  overflow: 'hidden',
  maxHeight: '0px',
  transition: `${common.transition}`,
  padding: `0px ${spaceMd}px`,
  backgroundColor: colours.offWhite,
  color: colours.greyDark
});

export const Content = styled.div({
  label: 'content',
  padding: `${common.paddingTB} 0px`
});
