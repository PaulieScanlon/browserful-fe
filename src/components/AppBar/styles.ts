import styled from 'react-emotion';

import { colours, common } from '../../theme';

export const AppBarWrapper = styled.header({
  label: 'app-bar-wrapper',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1,
  minHeight: common.appBarHeight
});
