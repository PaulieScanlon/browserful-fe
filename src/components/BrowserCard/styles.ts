import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  isExpanded?: boolean;
}

export const BrowserCardWrapper = styled.div({
  label: 'browser-card-wrapper',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1
});

export const BrowserCardHeader = styled.div<IProps>(
  {
    label: 'browser-card-header',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: `0 ${spaceMd}px`,
    padding: `${spaceMd}px 0px`,
    borderBottomWidth: '1px',
    borderBottomStyle: 'solid'
  },
  ({ isExpanded }) => ({
    borderBottomColor: isExpanded ? colours.greyLight : colours.white
  })
);

export const BrowserCardHeaderSection = styled.div({
  label: 'browser-card-header-section',
  display: 'inline-flex',
  alignItems: 'center'
});

export const BrowserCardBody = styled.div<IProps>(
  {
    label: 'browser-card-body',
    overflow: 'hidden'
  },
  ({ isExpanded }) => ({
    height: isExpanded ? '100%' : '0px',
    padding: isExpanded ? `${spaceMd}px ${spaceMd}px` : `0px ${spaceMd}px`
  })
);
