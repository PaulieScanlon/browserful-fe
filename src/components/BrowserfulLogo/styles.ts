import styled from 'react-emotion';

import { common, spaceLg } from '../../theme';

export const LogoWrapper = styled.div({
  label: 'logo-wrapper',
  display: 'inline-flex',
  alignItems: 'center',
  marginLeft: `${spaceLg}px`
});

export const SvgWrapper = styled.span({
  label: 'svg-wrapper',
  display: 'inline-flex',
  marginRight: '5px'
});
