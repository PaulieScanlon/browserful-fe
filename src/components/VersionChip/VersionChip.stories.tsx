import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';

import { VersionChip } from './VersionChip';

import browserslist from 'browserslist';
import { friendlyIfy } from '../../utils/friendlyIfy';

const mockData = friendlyIfy(browserslist(['last 1 Chrome versions']));

const stories = storiesOf('VersionChip', module);

const onChange = (event, browser, version) => {
  action('onChange')(event.currentTarget, browser, version);
};

stories.add(
  'default usage',
  withInfo('The browser and version prop are used on the onChange callback')(
    () => (
      <VersionChip
        browser={mockData[0].name}
        version={mockData[0].versions[0].id}
        onChange={(event, browser, version) =>
          onChange(event, browser, version)
        }
      />
    )
  )
);

stories.add(
  '!isIncluded & defaultChecked',
  withInfo('The defaultChecked prop is the checked status of the checkbox')(
    () => (
      <VersionChip
        defaultChecked={true}
        browser={mockData[0].name}
        version={mockData[0].versions[0].id}
        onChange={(event, browser, version) =>
          onChange(event, browser, version)
        }
      />
    )
  )
);

stories.add(
  'isIncluded',
  withInfo('The isIncluded prop shows if a browser/version is included')(() => (
    <VersionChip
      isIncluded={true}
      browser={mockData[0].name}
      version={mockData[0].versions[0].id}
      onChange={(event, browser, version) => onChange(event, browser, version)}
    />
  ))
);

stories.add(
  'isIncluded & defaultChecked',
  withInfo('The defaultChecked prop is the checked status of the checkbox')(
    () => (
      <VersionChip
        defaultChecked={true}
        isIncluded={true}
        browser={mockData[0].name}
        version={mockData[0].versions[0].id}
        onChange={(event, browser, version) =>
          onChange(event, browser, version)
        }
      />
    )
  )
);
