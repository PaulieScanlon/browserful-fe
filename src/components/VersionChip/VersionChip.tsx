import * as React from 'react';

import {
  VersionLabel,
  VersionText,
  VersionInput,
  VersionStyle
} from './styles';

interface IProps {
  browser: string;
  version: any;
  onChange: (
    event: React.ChangeEventHandler<HTMLInputElement>,
    browser: string,
    version: number | string
  ) => void;
  isIncluded?: boolean;
  defaultChecked?: boolean;
}

export const VersionChip: React.SFC<IProps> = ({
  version,
  browser,
  onChange,
  isIncluded,
  defaultChecked
}: IProps) => {
  return (
    <VersionLabel htmlFor={browser}>
      <VersionInput
        defaultChecked={defaultChecked}
        type="checkbox"
        id={browser}
        onChange={event => onChange(event as any, browser, version)}
        isIncluded={isIncluded}
      />
      <VersionStyle>
        <VersionText isIncluded={isIncluded}>{version || 0}</VersionText>
      </VersionStyle>
    </VersionLabel>
  );
};

VersionChip.defaultProps = {
  isIncluded: false,
  defaultChecked: false
};
