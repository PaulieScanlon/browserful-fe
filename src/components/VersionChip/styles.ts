import styled from 'react-emotion';

import { font } from '../../typography';
import { colours } from '../../theme';

interface IProps {
  isIncluded: boolean;
}

const getStyleRules = (isIncluded, checked) => {
  if (isIncluded) {
    return {
      borderColor: checked ? colours.green : colours.transparent,
      backgroundColor: colours.greenLight
    };
  } else {
    return {
      borderColor: checked ? colours.red : colours.transparent,
      backgroundColor: colours.redLight
    };
  }
};

export const VersionLabel = styled.label({
  label: 'version-label',
  position: 'relative',
  display: 'inline-flex',
  cursor: 'pointer',
  justifyContent: 'center',
  alignItems: 'center'
});

export const VersionText = styled.span<IProps>(
  {
    fontSize: '12px',
    lineHeight: '14px',
    position: 'absolute',
    fontFamily: `${font.fontFamily}`,
    textAlign: 'center',
    whiteSpace: 'pre-wrap',
    textTransform: 'capitalize'
  },
  ({ isIncluded }) => ({
    fontWeight: isIncluded ? 'bold' : 'normal',
    color: isIncluded ? colours.green : colours.red
  })
);

export const VersionInput = styled.input<IProps>(
  {
    label: 'verion-input',
    position: 'absolute',
    display: 'none'
  },
  ({ isIncluded }: IProps) => ({
    ':checked + span': {
      ...getStyleRules(isIncluded, true)
    },
    '+ span': {
      ...getStyleRules(isIncluded, false)
    }
  })
);

export const VersionStyle = styled.span({
  label: 'version-style',
  width: '40px',
  height: '40px',
  borderRadius: '20px',
  borderStyle: 'solid',
  boxSizing: 'border-box',
  outline: 'none'
});
