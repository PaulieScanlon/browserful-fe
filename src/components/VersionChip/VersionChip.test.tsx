import * as React from 'react';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import { colours } from '../../theme';

expect.extend(createMatchers(emotion));

import { shallow, mount } from 'enzyme';

import { ChipButton } from './styles';
import { VersionChip } from './VersionChip';

const onClick = (event, id) => {};

const shallowDefault = props => {
  return shallow(
    <VersionChip
      browser="Chrome"
      version={1}
      onClick={(event, id) => onClick(event, id)}
      {...props}
    />
  );
};

const mountDefault = props => {
  return mount(
    <VersionChip
      browser="Chrome"
      version={1}
      onClick={(event, id) => onClick(event, id)}
      {...props}
    />
  );
};

describe('<VersionChip />', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({ variant: 'included' });
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('calls function onClick', () => {
    const testOnClick = jest.fn();
    const wrapper = shallow(
      <VersionChip
        browser="Chrome"
        version={1}
        variant="included"
        onClick={(event, id) => testOnClick(event, id)}
      />
    );
    const chipButton = wrapper.find(ChipButton);
    chipButton.simulate('click');
    expect(testOnClick).toHaveBeenCalledTimes(1);
  });
  it('has a browser prop', () => {
    const wrapper = mountDefault({ variant: 'included' });
    expect(wrapper.props().browser).toEqual('Chrome');
  });
  it('has a version prop', () => {
    const wrapper = mountDefault({ variant: 'included' });
    expect(wrapper.props().version).toEqual(1);
  });
  it('has a variant prop', () => {
    const wrapper = mountDefault({ variant: 'included' });
    expect(wrapper.props().variant).toEqual('included');
  });
  it('has a chipButton', () => {
    const wrapper = mountDefault({ variant: 'included' });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveLength(1);
  });
  it('has correct styles for included', () => {
    const wrapper = mountDefault({ variant: 'included' });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveStyleRule(
      'background-color',
      `${colours.greenLight}`
    );
  });
  it('has correct styles for excluded', () => {
    const wrapper = mountDefault({ variant: 'excluded' });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveStyleRule(
      'background-color',
      `${colours.redLight}`
    );
  });
  it('has correct border colour for included isSelected true', () => {
    const wrapper = mountDefault({ variant: 'included', isSelected: true });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveStyleRule('border-color', `${colours.green}`);
  });
  it('has correct border colour for excluded isSelected true', () => {
    const wrapper = mountDefault({ variant: 'excluded', isSelected: true });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveStyleRule('border-color', `${colours.red}`);
  });
  it('has correct border-width', () => {
    const wrapper = mountDefault({ variant: 'included', isSelected: true });
    const chipButton = wrapper.find(ChipButton);
    expect(chipButton).toHaveStyleRule('border-width', '2px');
  });
});
