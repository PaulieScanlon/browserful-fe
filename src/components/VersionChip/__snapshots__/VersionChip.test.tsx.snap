// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VersionChip /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VersionChip
    browser="Chrome"
    defaultChecked={false}
    isIncluded={false}
    onChange={[Function]}
    version={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <VersionInput
          defaultChecked={false}
          id="Chrome"
          isIncluded={false}
          onChange={[Function]}
          type="checkbox"
        />,
        <VersionStyle />,
        <VersionText
          isIncluded={false}
        >
          1
        </VersionText>,
      ],
      "htmlFor": "Chrome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultChecked": false,
          "id": "Chrome",
          "isIncluded": false,
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 1,
          "isIncluded": false,
        },
        "ref": null,
        "rendered": 1,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <VersionInput
            defaultChecked={false}
            id="Chrome"
            isIncluded={false}
            onChange={[Function]}
            type="checkbox"
          />,
          <VersionStyle />,
          <VersionText
            isIncluded={false}
          >
            1
          </VersionText>,
        ],
        "htmlFor": "Chrome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultChecked": false,
            "id": "Chrome",
            "isIncluded": false,
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 1,
            "isIncluded": false,
          },
          "ref": null,
          "rendered": 1,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
